### PR TITLE
Bookstore: add reducers and actions

### PR DESCRIPTION
**Pull request by RaoAkif**
- Write your book's actions and reducer (in /src/redux/books/books.js) using the ducks pattern.
  - Define action types for adding and removing a book.
  - Set the initial state to be an empty array of books.
  - Export Action Creators for your actions.
  - Write your reducer and export it as default.
      - Define state changes for the actions that you created.
      - In case of unknown action - return the current state.

- Write your categories actions and reducer (in /src/redux/categories/categories.js) using the ducks pattern.
  - Define an action type for checking the status.
  - Set the initial state to be an empty array of categories.
  - Export Action Creators for your actions.
  - Write your reducer and export it as default.
    - For the check status action return a string "Under construction".
    - In case of unknown action - return the current state.

- Configure the Redux Store (/src/redux/configureStore.js).
  - Importing the necessary methods from Redux.
  - Combine both reducers into a root reducer by using combineReducers function.
  - Use the root reducer to create your store.

### CODE REVIEW

File: src/redux/books/books.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  }
+  return result;
+}
+
+const uuid = makeIdWithLength(20);
+// console.log(uuid);
+
+//
+const ADD_BOOK = 'ADD_BOOK';
+const DELETE_BOOK = 'DELETE_BOOK';
```
Comment (FrederickMih) [inline src/redux/books/books.js]: - [ ] Please let's make sure to always scope action types based on the feature they represent. This is very important especially if you have plans to scale your app in the future. You will be required to do something like this: 
 
  `const ADD_BOOK = 'bookstore/books/ADD_BOOK';`
   
  This is referred to as the [ducks pattern](https://github.com/erikras/ducks-modular-redux)

File: src/redux/categories/categories.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
+const CHECK_STATUS = 'CHECK_STATUS';
```
Comment (FrederickMih) [inline src/redux/categories/categories.js]: - [ ] As already mentioned above, you would need to scope your action types in regards to the feature they represent. Let's fix this issue, please.

**Review by FrederickMih — CHANGES_REQUESTED** [review]
Hi @RaoAkif , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


# To Highlights

:fast_forward: Linters checks are all passing :heavy_check_mark: 
:fast_forward: GitHub workflow has been followed :heavy_check_mark: 
:fast_forward: Pull Request has a title and summary :heavy_check_mark: 

## Required Changes ♻️

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/books/books.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  }
+  return result;
+}
+
+const uuid = makeIdWithLength(20);
+// console.log(uuid);
+
+//
+const ADD_BOOK = 'bookstore/books/ADD_BOOK';
+const DELETE_BOOK = 'bookstore/books/DELETE_BOOK';
```
Comment (mateo951) [inline src/redux/books/books.js:17]: Addressed ✔

File: src/redux/categories/categories.js [hunk 4]
```diff
@@ -0,0 +1,18 @@
+const CHECK_STATUS = 'bookstore/books/CHECK_STATUS';
```
Comment (mateo951) [inline src/redux/categories/categories.js]: Addressed ✔

**Review by mateo951 — APPROVED** [review]
### Hi @RaoAkif, 

Your project is complete!  There is nothing else to say other than... it's time to merge it 🔥 
### Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**